cmd/upload.go

```diff
@@ -0,0 +1,112 @@
+/**
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Privado-Inc/privado-cli/pkg/config"
+	"github.com/Privado-Inc/privado-cli/pkg/docker"
+	"github.com/Privado-Inc/privado-cli/pkg/fileutils"
+	"github.com/Privado-Inc/privado-cli/pkg/telemetry"
+	"github.com/spf13/cobra"
+)
+
+var uploadCmd = &cobra.Command{
+	Use:   "upload <repository>",
+	Short: "Sync the results with the privado.ai Cloud Dashboard",
+	Args:  cobra.ExactArgs(1),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		telemetry.DefaultInstance.RecordAtomicMetric("version", Version)
+		telemetry.DefaultInstance.RecordAtomicMetric("cmd", strings.Join(os.Args, " "))
+	},
+	Run: upload,
+	PostRun: func(cmd *cobra.Command, args []string) {
+		telemetryPostRun(nil)
+	},
+}
+
+func upload(cmd *cobra.Command, args []string) {
+	repository := args[0]
+	debug, _ := cmd.Flags().GetBool("debug")
+	// overwriteResults, _ := cmd.Flags().GetBool("overwrite")
+
+	hasUpdate, updateMessage, err := checkForUpdate()
+	if err == nil && hasUpdate {
+		fmt.Println(updateMessage)
+		time.Sleep(config.AppConfig.SlowdownTime)
+		fmt.Println("To use the latest version of Privado CLI, run `privado update`")
+		time.Sleep(config.AppConfig.SlowdownTime)
+		fmt.Println()
+	}
+
+	if dockerAccessKey, err := docker.GetPrivadoDockerAccessKey(true); err != nil || dockerAccessKey == "" {
+		exit(fmt.Sprintf("Cannot fetch docker access key: %v \nPlease try again or raise an issue at %s", err, config.AppConfig.PrivadoRepository), true)
+	} else {
+		config.LoadUserDockerHash(dockerAccessKey)
+	}
+
+	// "always pass -ic: even when internal rules are ignored (-i)"
+	commandArgs := []string{
+		config.AppConfig.Container.SourceCodeVolumeDir,
+	}
+
+	entrypoint := []string{
+		config.AppConfig.Container.PrivadoCoreBinPath, "upload",
+	}
```

Comment:
Refactor comments: 
1. Can we switch this order (entrypoint first)? Will make it easier for new contributors to relate.
2. We can also remove comments which are not required anymore.
3. Also, check the indent, looks a little weird. VSCode should automatically do it on save if you have the default Go extensions installed & configured

Reply:
Indent was VSCode automatic. I'll change the ordering and comments.